plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessor.java

```diff
@@ -100,36 +100,25 @@ protected SR createFileSR(final Connection conn, final String path) {
         PBD pbd = null;
 
         try {
-            final String srname = hypervisorResource.getHost().getUuid() + path.trim();
-
-            final Set<SR> srs = SR.getByNameLabel(conn, srname);
-
-            if (srs != null && !srs.isEmpty()) {
-                return srs.iterator().next();
+            final String srname = path.trim();
```

Comment:
@rhtyd, since sr is unique on a pool, adding hostname to its path does not give any advantage. Usually MS will go thru the pool master so effectively it will end up the pool master uuid to sr name.
This could have avoided the race condition by probably introducing redundant SRs, but now that the race condition is dealt with we really donot need the host uuid qualification of SR name.